esp/templates/program/modules/studentclassregmodule/catalog.html

```diff
@@ -108,166 +116,6 @@ <h2 style="text-align: center;">
-        .filter(gradeFilter(grade))
-        .filter(difficultyFilter(difficulty))
-        .filter(openFilter(status))
-        .filter(durationFilter(duration))
-
-    classes.forEach(hideClass)
-    shownClasses.forEach(showClass)
-}
-
-function getOptions() {
-    return {
-        grade: document.getElementById("grade_filter").value,
-        difficulty: document.getElementById("difficulty_filter").value,
-        status: document.getElementById("status_filter").value,
-        duration: document.getElementById("duration_filter").value,
-    }
-}
-
-var student_grade = esp_user.cur_grade;
-if (student_grade != "" && student_grade != null) {
-    student_grade = parseInt(student_grade);
-    const gradeFilterElem = document.getElementById("grade_filter")
-    if (gradeFilterElem.querySelector(`option[value="${student_grade}"]`)) {
-        gradeFilterElem.value = student_grade
-    }
-}
-
-applyCurrentFilters()
-
-// Set up duration options by finding all unique durations from the classes in the catalog
-const durationFilterElem = document.getElementById("duration_filter")
-let uniqueLengths = [
-        // Convert to and back from a Set to keep only unique values
-        ...new Set(
-            Array.from(document.getElementsByClassName("show_class"))
-                .map(cls=>cls.getAttribute("data-duration"))
-        )
-    ]
-    .sort((a,b)=>a-b) // least to greatest
-    .map((val)=>[val, floatToFormattedTime(val)])
-    .forEach(([raw, formatted])=>{
-        let option = document.createElement("option")
-        option.setAttribute("value", raw)
-        option.textContent = formatted
-        durationFilterElem.append(option)
-    })
-
-// Set up difficulties by getting them from tag data (injected further up)
-const difficultyFilterElem = document.getElementById("difficulty_filter")
-
-DIFFICULTIES.forEach(([difficulty])=>{
-    let option = document.createElement("option")
-    option.setAttribute("value", difficulty)
-    option.textContent = difficulty
-    difficultyFilterElem.append(option)
-})
-
-function setModifiedColor(selectElem) {
-    if (selectElem.value !== "all") {
-        selectElem.style.background = MODIFIED_COLOR
-    } else {
-        selectElem.style.background = ""
-    }
-}
-
-FILTER_IDS.forEach((id)=>{
-    const filterElem = document.getElementById(id)
-    // Covers the case when a page refresh has preserved form values
-    // or the grade is automatically set.
-    setModifiedColor(filterElem)
-
-    filterElem.addEventListener("change", ()=>{
-        setModifiedColor(filterElem)
-        applyCurrentFilters()
-    })
-})
-</script>
-
-
-<script type="text/javascript">
-function configure_addbuttons()
-{
-    {% if register_from_catalog %}
-    //  Registration from the catalog is allowed
-    $j("input.addbutton").show();
-    {% endif %}
-}
-$j(document).ready(configure_addbuttons);
-</script>
+<script type="text/javascript" src="/media/scripts/program/modules/catalog.js"></script>
```

Comment:
Good call, I should have done this in my PR because this is prob too much JS to inline.